pkg/publicshare/publicshare.go

```diff
@@ -100,6 +100,8 @@ func MatchesFilter(share *link.PublicShare, filter *link.ListPublicSharesRequest
 	switch filter.Type {
 	case link.ListPublicSharesRequest_Filter_TYPE_RESOURCE_ID:
 		return utils.ResourceIDEqual(share.ResourceId, filter.GetResourceId())
+	case link.ListPublicSharesRequest_Filter_Type(4):
+		return share.ResourceId.StorageId == filter.GetResourceId().GetStorageId()
```

Comment:
Should we move `PublicShareSpaceFilterType` and `ShareSpaceFilterType` to `utils` package so we can use it here too? Or we don't mind?

Reply:
So I refactored it in another way. Now the constants are only in the `share` and `publicshare` package.